tool/input_sdk/private/isolate_helper.dart

```diff
@@ -697,12 +697,14 @@ class _IsolateEvent {
 class _MainManagerStub {
   void postMessage(msg) {
     // "self" is a way to refer to the global context object that
-    // works in HTML pages and in Web Workers.  It does not work in d8
-    // and Firefox jsshell, because that would have been too easy.
+    // works in HTML pages and in Web Workers.  It does not work in d8, iojs
+    // and Firefox jsshell, because that would have been too easy. In iojs
+    // "global" works.
     //
     // See: http://www.w3.org/TR/workers/#the-global-scope
     // and: http://www.w3.org/TR/Window/#dfn-self-attribute
-    JS("void", r"self.postMessage(#)", msg);
+    JS("void",
+       r"(typeof global == 'undefined' ? self : global).postMessage(#)", msg);
```

Comment:
should probably save this into a variable and use it.

final _global = JS('...');

elsewhere:

    return JS('', '#.postMessage(#)', _global, msg);

if we can find a way to fix `dart.global` that makes it even easier:

    return JS('', 'dart.global.postMessage(#)', _global, msg);